### PR TITLE
Implement Sqlite3 Module

### PR DESCRIPTION
**Pull request by qingshi163**

### CODE REVIEW

**Comment by qingshi163** [comment]
@youknowone Can you help with that compilation fail with android?

**Comment by youknowone** [comment]
that looks like a toolchain problem. rather than fighting with it, how about just disabling sqlite3 for android?

File: stdlib/src/sqlite.rs [hunk 1]
```diff
@@ -0,0 +1,2963 @@
...
+            if guard.is_some() {
+                Ok(PyMutexGuard::map(guard, |x| unsafe {
+                    x.as_mut().unwrap_unchecked()
+                }))
+            } else {
+                Err(new_programming_error(
+                    vm,
+                    "Cannot operate on a closed cursor.".to_owned(),
+                ))
+            }
```
Comment (youknowone) [inline stdlib/src/sqlite.rs:1375]:
```suggestion
            let Some(guard) = self.inner.lock() else {
                return Err(new_programming_error(
                    vm,
                    "Cannot operate on a closed cursor.".to_owned(),
                ));
            };
            Ok(PyMutexGuard::map(guard, |x| unsafe {
                x.as_mut().unwrap_unchecked()
            }))
```

File: stdlib/src/sqlite.rs [hunk 2]
```diff
@@ -0,0 +1,2963 @@
...
+            zelf
+        }
+
+        #[pymethod(magic)]
+        fn exit(&self, _args: FuncArgs) {
+            self.close()
+        }
+
+        fn inner(&self, vm: &VirtualMachine) -> PyResult<PyMappedMutexGuard<BlobInner>> {
+            let guard = self.inner.lock();
```
Comment (youknowone) [inline stdlib/src/sqlite.rs:2032]: let-else for guard here

**Review by youknowone — APPROVED** [review]
Great! You made a huge work!

File: stdlib/src/sqlite.rs [hunk 3]
```diff
@@ -0,0 +1,2963 @@
...
+            if guard.is_some() {
+                Ok(PyMutexGuard::map(guard, |x| unsafe {
+                    x.as_mut().unwrap_unchecked()
+                }))
+            } else {
+                Err(new_programming_error(
+                    vm,
+                    "Cannot operate on a closed database.".to_owned(),
+                ))
+            }
```
[the remark this answers is not in the document]
Comment (qingshi163) [reply stdlib/src/sqlite.rs:856]: This is not going to work as the variable guard is MutexGuard<Option<_>>

File: stdlib/src/sqlite.rs (hunk 2 above)
Comment (fanninpm) [reply stdlib/src/sqlite.rs:2028]: Can you do let-else on a `MutexGuard`?

**Pull request merged**